Modules/termios.c

```diff
@@ -212,17 +212,28 @@ termios_tcsetattr_impl(PyObject *module, int fd, int when, PyObject *term)
         return PyErr_SetFromErrno(state->TermiosError);
     }
 
-    mode.c_iflag = (tcflag_t) PyLong_AsLong(PyList_GetItem(term, 0));
-    mode.c_oflag = (tcflag_t) PyLong_AsLong(PyList_GetItem(term, 1));
-    mode.c_cflag = (tcflag_t) PyLong_AsLong(PyList_GetItem(term, 2));
-    mode.c_lflag = (tcflag_t) PyLong_AsLong(PyList_GetItem(term, 3));
-    speed_t ispeed = (speed_t) PyLong_AsLong(PyList_GetItem(term, 4));
-    speed_t ospeed = (speed_t) PyLong_AsLong(PyList_GetItem(term, 5));
-    PyObject *cc = PyList_GetItem(term, 6);
-    if (PyErr_Occurred()) {
-        return NULL;
+    long item;
+#define SAFE_LONG_ITEM(obj) \
+    item = PyLong_AsLong(obj); \
+    if (item == -1 && PyErr_Occurred()) { \
+        return NULL; \
     }
 
+    SAFE_LONG_ITEM(PyList_GET_ITEM(term, 0));
+    mode.c_iflag = (tcflag_t) item;
+    SAFE_LONG_ITEM(PyList_GET_ITEM(term, 1));
+    mode.c_oflag = (tcflag_t) item;
+    SAFE_LONG_ITEM(PyList_GET_ITEM(term, 2));
+    mode.c_cflag = (tcflag_t) item;
+    SAFE_LONG_ITEM(PyList_GET_ITEM(term, 3));
+    mode.c_lflag = (tcflag_t) item;
+    SAFE_LONG_ITEM(PyList_GET_ITEM(term, 4));
+    speed_t ispeed = (speed_t) item;
+    SAFE_LONG_ITEM(PyList_GET_ITEM(term, 5));
+    speed_t ospeed = (speed_t) item;
+#undef SAFE_LONG_ITEM
```

Comment:
I would consider to specialise the macro even more:

```suggestion
#define SET_FROM_LIST(TYPE, VAR, LIST, N) do {  \
    PyObject *item = PyList_GET_ITEM(LIST, N);  \
    long num = PyLong_AsLong(item);             \
    if (num == -1 && PyErr_Occurred()) {        \
        return NULL;                            \
    }                                           \
    VAR = (TYPE)num;                            \
} while (0)
    SET_FROM_LIST(tcflag_t, mode.c_iflag, term, 0);
    SET_FROM_LIST(tcflag_t, mode.c_oflag, term, 1);
    SET_FROM_LIST(tcflag_t, mode.c_cflag, term, 2);
    SET_FROM_LIST(tcflag_t, mode.c_lflag, term, 3);
    SET_FROM_LIST(speed_t, ispeed, term, 4);
    SET_FROM_LIST(speed_t, ospeed, term, 5);
#undef SET_FROM_LIST
```

Reply:
I had a similar thought, but then my Rust background didn't allow me to overuse macros :)

Reply:
I wish I had a Rust background!

Reply:
My suggestion had bugs! 😆

Reply:
I will fix them shortly. 

> I wish I had a Rust background!

https://github.com/RustPython/RustPython 👀

Reply:
A simple fix can be to simply declare the variables above the macro definition.